tests/docs/testing_framework.md

```diff
@@ -0,0 +1,97 @@
+
+To enable custom integration testing for any of the cases above, a test class must be
+written which inherits from `CustomIntegrationTest` under tests/custom_test. Within this
+class, two paths can be defined: `custom_scripts_directory` which points to the
+directory containing all the custom scripts which are to be tested and 
+`custom_class_directory` which points to the directory containing all the custom test
+classes.
+
+Similar to the non-custom integration testing, each custom integration test script or
+test class must include a .yaml file which includes 3 values 
+(defined by the `CustomTestConfig` dataclass found under `tests/data`): 
+`test_type` which indicates if the test is a sanity, smoke or regression test, 
+`cadence`: which dictates how often the test case runs (on commit, weekly, or nightly), 
+and the `script_path` which lists the name of the custom testing script or custom test
+class within the directory. 
+
+An example of an implementation of the `CustomIntegrationTest` can be found under
+`tests/examples`
+
+## Additional markers and decorators 
+
+- New markers are added in to mark tests as `unit`, `integration`, `smoke`, and `custom` 
+tests allowing us to run a subset of the tests when needed
+- Two new decorators are added in to check for package and compute requirements. If
+the requirements are not met, the test is skipped. Currently, `requires_torch` and 
+`requires_gpu` are added in and can be found under `testing_utils.py`
+
+## Testing Targets 
+
+### Unit Testing Targets:
+- A unit test should be written for every utils, helper, or static function. 
+    - Test cases should be written for all datatype combinations that the function takes as input 
+    - Can have `smoke` tests but focus should be on `sanity`
+
+### Integration Testing Targets:
+- An integration test should be written for every cli pathway that is exposed through `setup.py`
+    - All cli-arg combinations should be tested through a `smoke` check 
+    (all may be overkill but ideally we're covering beyond the few important combinations)
+    - All **important** cli-arg combinations should be covered through either a `sanity`
+    check or a `regression` check
+        - A small model should be tested through a `sanity` check  
+        - All other larger models should be tested through `regression` test types
+   
+- An integration test should be written for every major/critical module 
+    - All arg combinations should be tested through a `smoke` check
+    (all may be overkill but ideally we're covering beyond the few important combinations)
+    - All **important** arg combinations should be covered through either a `sanity`
+    check or a `regression` check
+        - A small model should be tested through a `sanity` check  
+        - All other larger models should be tested through `regression` test types
+
+## End-to-end Testing Targets:
+- Tests cascading repositories (sparseml --> vLLM) but will become more prominent as are
```

Comment:
Perhaps add that there should be at least one use case scenario that emulates how a user is expected to use a particular feature being tested.